src/Bonsai.SpikeGLX/DigitalOutput.cs

```diff
@@ -38,39 +38,42 @@ public class DigitalOutput
 
         /// <summary>
         /// Sets one or more digital output lines through SpikeGLX from an observable sequence
-        /// of integer values representing a digital state.
+        /// of boolean values representing a digital state.
         /// </summary>
         /// <param name="source">
-        /// A sequence of 32-bit signed integers, where each value represents a high (1) or low (0)
-        /// state to which to set the digital output line.
+        /// A sequence of boolean values, where each value represents a high (<see langword="true"/>)
+        /// or low (<see langword="false"/>) state to which to set the digital output line.
         /// </param>
         /// <returns>
         /// An observable sequence that is identical to the <paramref name="source"/> sequence
         /// but where there is an additional side effect of setting digital output lines through
         /// SpikeGLX.
         /// </returns>
-        public IObservable<int> Process(IObservable<int> source)
+        public IObservable<bool> Process(IObservable<bool> source)
         {
             return Observable.Using(() => new SpikeGLX(Host, Port),
-                connection => source.Do(input => connection.SetDigitalOut(input, Channels)));
+                connection => source.Do(input => connection.SetNIDigitalOut(Channels,
+                input ? uint.MaxValue : uint.MinValue)));
         }
 
         /// <summary>
         /// Sets one or more digital output lines through SpikeGLX from an observable sequence
-        /// of boolean values representing a digital state.
+        /// of integer values representing a digital state.
         /// </summary>
         /// <param name="source">
-        /// A sequence of boolean values, where each value represents a high (<see langword="true"/>)
-        /// or low (<see langword="false"/>) state to which to set the digital output line.
+        /// A sequence of 32-bit signed integers, where each value represents a high (1) or low (0)
+        /// state to which to set the digital output line.
         /// </param>
         /// <returns>
         /// An observable sequence that is identical to the <paramref name="source"/> sequence
         /// but where there is an additional side effect of setting digital output lines through
         /// SpikeGLX.
         /// </returns>
-        public IObservable<int> Process(IObservable<bool> source)
+        public IObservable<int> Process(IObservable<int> source)
         {
-            return Process(source.Select(input => input ? 1 : 0));
+            return Observable.Using(() => new SpikeGLX(Host, Port),
+                connection => source.Do(input => connection.SetNIDigitalOut(Channels,
+                input != 0 ? uint.MaxValue : uint.MinValue)));
```

Comment:
@J-M-White I was wondering whether here we might want to simply pass on the unmodified `int` value itself, so we afford the possibility of setting the state of individual channels according to the bitmask?

In fact, it might be easier to simply make the overload take a `uint` directly so that it aligns exactly with the SpikeGLX API.

In order to reduce the git diff from the current version, you might also want to swap the order of the overloads (`uint` first, then the `bool`) so that it becomes easier to spot the relevant differences.